src/engraving/rendering/score/horizontalspacing.cpp

```diff
@@ -1034,7 +1034,9 @@ void HorizontalSpacing::setPositionsAndWidths(const std::vector<SegmentPosition>
         Measure* curSegMeasure = curSeg->measure();
         Measure* nextSegMeasure = nextSeg->measure();
 
-        double segWidth = curSegMeasure == nextSegMeasure || nextSeg->isStartRepeatBarLineType() ? nextX - curX : curSeg->minRight();
+        double segWidth = curSegMeasure == nextSegMeasure || nextSeg->isStartRepeatBarLineType()
+                          || nextSeg->isKeySigType() || nextSeg->isTimeSigType()
+                          || nextSeg->isClefType() ? nextX - curX : curSeg->minRight();

```

Comment:
This is unreadable now though. Perhaps a `bool computeWidthByDifferenceFromNext` or something like that could help